IS0801Test.py

```diff
@@ -12,12 +12,405 @@
+        for outputInstance in outputList:
+            sourceID = outputInstance.getSourceID()
+            for inputInstance in inputList:
+                inputParent = inputInstance.getParent()
+                if sourceID == inputParent:
+                    route = {
+                        "input": inputInstance,
+                        "output": outputInstance
+                    }
+                    forbiddenRoutes.append(route)
+
+        for route in forbiddenRoutes:
+            outputCaps = route['output'].getCaps()
+            msg = ("It is possible to create a loop using re-entrant matricies" 
+                   " between input {} and output {}".format(route['input'].id, route['output'].id))
+            try:
+                routableInputs = outputCaps['routable_inputs']
+            except KeyError:                
+                return test.FAIL(msg)
+            if route['output'].id not in routableInputs:
+                return test.FAIL(msg)
+        return test.PASS()
+
+    def test_09(self):
+        """Human readable name provided in the props resource"""
+        test = Test("Check for human readable name provided in the props resource.")
+        return test.MANUAL()
+
+    def test_10(self):
+        """Human readable description provided in the props resource"""
+        test = Test("Check for human readable description provided in the props resource.")
+        return test.MANUAL()
+        pass
+
+    def test_11(self):
+        """Inputs have at least one channel represented in their channels resource"""
+        test = Test("Inputs have at least one channel represented in their channels resource")
+        globalConfig.test = test
+        inputList = getInputList()
+        for inputInstance in inputList:
+            channels = inputInstance.getChannelList()
+            if len(channels) == 0:
+                return test.FAIL("Inputs must have at least one channel")
+        return test.PASS()
+
+    def test_12(self):
+        """Outputs have at least one channel represented in their channels resource"""
+        test = Test("Outputs have at least one channel represented in their channels resource")
+        globalConfig.test = test
+
+        outputList = getOutputList()
+        for outputInstance in outputList:
+            channels = outputInstance.getChannelList()
+            if len(channels) == 0:
+                return test.FAIL("Outputs must have at least one channel")
+        return test.PASS()
+
+    def test_13(self):
+        """Attempting to violate routing constraints results in an HTTP 400 response"""
+        test = Test("Attempting to violate routing constraints results in an HTTP 400 response")
+        globalConfig.test = test
+
+        outputList = getOutputList()
+        constrainedOutputList = []
+        for outputInstance in outputList:
+            constraints = outputInstance.getCaps()
+            try:
+                routableInputs = constraints['routeable_inputs']

```

Comment:
This may be in the spec and so need correcting there instead, but I think it should be 'routable_inputs'.

Reply:
Fine in the spec - it's just Riedel's instance I've been testing against doesn't have any of this constraint so it didn't get run!